libkbfs/block_types.go

```diff
@@ -81,27 +80,25 @@ func (cb *CommonBlock) NewEmpty() Block {
 	return NewCommonBlock()
 }
 
+// ToCommonBlock implements the Block interface for CommonBlock.
+func (cb *CommonBlock) ToCommonBlock() *CommonBlock {
+	return cb
+}
+
 // Set implements the Block interface for CommonBlock.
-func (cb *CommonBlock) Set(other Block, codec kbfscodec.Codec) {
-	// Don't assert type for CommonBlock because we need to be able to Set from
-	// specific block types
-	err := kbfscodec.Update(codec, cb, other)
-	if err != nil {
-		panic("Unable to CommonBlock.Set")
-	}
-	cb.cacheMtx.Lock()
-	defer cb.cacheMtx.Unlock()
-	cb.cachedEncodedSize = other.GetEncodedSize()
+func (cb *CommonBlock) Set(other Block) {
+	otherCopy := other.ToCommonBlock()
+	cb.IsInd = otherCopy.IsInd
+	cb.UnknownFieldSetHandler = otherCopy.UnknownFieldSetHandler
+	cb.SetEncodedSize(otherCopy.GetEncodedSize())
 }
 
-// Copy copies a CommonBlock without the lock.
-func (cb *CommonBlock) Copy() CommonBlock {
-	cb.cacheMtx.RLock()
-	defer cb.cacheMtx.RUnlock()
+// DeepCopy copies a CommonBlock without the lock.
+func (cb *CommonBlock) DeepCopy() CommonBlock {
 	return CommonBlock{
 		IsInd: cb.IsInd,
 		UnknownFieldSetHandler: cb.UnknownFieldSetHandler,
```

Comment:
Weren't you going to add a deep copy for the unknown fields too?  Obviously this is fine since we did it before too, but for some reason the name change from Copy -> DeepCopy makes me feel like this should be deeply-copied too.